### ISSUE TITLE
Are there plans to update this to be compatible with newer pipewire?

### ISSUE THREAD

**Issue by Quackdoc** [EHfive/pw-capture#2]
Pipewire has gone through a lot of changes and this no longer compiles due to libspa issues.

**Comment by EHfive** [comment]
Adapted the client to latest pipewire-rs in d8419d42ac1c45d098e80963b72d34fbcc3ffdcb, this should now compiles against PipeWire 1.0 again.

This project is sort of abandoned as I did not use it very often personally, so I would not guarantee future updates on it.

Also with transition to Wayland, which also provides DMA-BUF window capture, the `pw-capture` layer is becoming redundant.

**Comment by Quackdoc** [comment]
Unforunately I wasn't able to get it to work due to a rather puzzling issue Im not sure if is related to this program, pipewire-rs or pipewire itself, Thanks for updating it though it will help me a lot when I figure out how to get it working, I am currently on wayland now but cosmic doesn't really offer what I need but this looks like it will work great onec I get it working.


```ps
[pw-capture-vk] DEBUG [client/src/client.rs:140] [pw_capture_client::client] pw-thread
[D][25180.664455] pw.context   | [      pipewire.c:  236 load_spa_handle()] load lib:'support/libspa-support' factory-name:'support.cpu'
[D][25180.664525] spa.cpu      | [           cpu.c:  256 impl_init()] 0x76140c008168: count:12 align:32 flags:0007cbdb
[I][25180.664541] pw.context   | [      pipewire.c:  579 pw_init()] version 1.1.0
[D] pw.context [pipewire.c:195 unref_handle()] clear handle 'support.cpu'
[D] pw.context [pipewire.c:195 unref_handle()] clear handle 'support.log'
[D] pw.context [pipewire.c:156 unref_plugin()] unloaded plugin:'/usr/lib/spa-0.2/support/libspa-journal.so'
[D] pw.context [pipewire.c:195 unref_handle()] clear handle 'support.log'
[D] pw.context [pipewire.c:156 unref_plugin()] unloaded plugin:'/usr/lib/spa-0.2/support/libspa-support.so'
[D] pw.context [pipewire.c:236 load_spa_handle()] load lib:'(null)' factory-name:'support.system'
[E] pw.context [pipewire.c:242 load_spa_handle()] load lib: plugin directory undefined, set SPA_PLUGIN_DIR
[E] pw.loop [loop.c:69 pw_loop_new()] 0x76140c006320: can't make support.system handle: No such file or directory
[pw-capture-vk] ERROR [vulkan/src/lib.rs:115] [client init] failed to create client: failed to connect PipeWire Server

Caused by:
    receiving on an empty and disconnected channel
[pw-capture-vk] ERROR [vulkan/src/lib.rs:1339] [create_swapchain_khr] failed to create stream: failed to get client
[pw-capture-vk] DEBUG [vulkan/src/lib.rs:1302] [create_swapchain_khr] created: 0x5bb33787f760, old: 0x5bb3378c24f0
[pw-capture-vk]  INFO [vulkan/src/lib.rs:1155] [create_stream] creating stream, extent: 666x700 format: VkFormatInfo { format: BGRA, transfer: UNORM, vk_format: B8G8R8A8_UNORM }

```

**Comment by EHfive** [comment]
> Unforunately I wasn't able to get it to work due to a rather puzzling issue Im not sure if is related to this program

This is a regression introduced in 058d232ce93631ba54378ea021888cf22dd3ed15, fixed in latest 2cafa30430c01a56487613d4ba59bd0ef9743679. Sorry for inconvenience.
